### ISSUE TITLE
zero loss

### ISSUE THREAD

**Issue by ShangwuYao** [SeanNaren/warp-ctc#59]
I am getting zero loss in calculating the CTCLoss, I don't know if it is supposed to happen.
logits, labels, input_length, labels_length could be found in
https://github.com/ShangwuYao/support-issues/tree/master/warpctc

**Comment by Jackson-Kang** [comment]
Same error, too.. 

Did you solved the problem?

**Comment by guilk** [comment]
@ShangwuYao Did you solve this problem? My training loss does not decrease and all predictions are _blank characters.

**Comment by ShangwuYao** [comment]
No, not yet

**Comment by Jackson-Kang** [comment]
Try to run the example written in README.md. Is loss still 0?

**Comment by ShangwuYao** [comment]
No the example works fine.

**Comment by FAhmedB** [comment]
This happened to me when the number of features is less than the number of labels

**Comment by simpleoier** [comment]
It seems that I also met the same case with @FAhmedB , when the seq-length of features is less than the seq-length of labels.

**Comment by MarStarck** [comment]
same problem.
but if don't use cuda the loss is normal. so I think it may be environment problem. I'm using cuda8.0, pytorch 0.4.1

**Comment by MarStarck** [comment]
I installed cuda 9.0 with cudnn 7, pytorch 0.4.0 and remake and reinstall it and it is not 0 anymore!

**Comment by jordanott** [comment]
> @ShangwuYao Did you solve this problem? My training loss does not decrease and all predictions are _blank characters.

My training loss fluctuates but all predictions are blank characters. 
Any suggestions?

**Comment by pxkc** [comment]
I've installed pytorch0.4.0  instead of the previous 0.2.0. And reinstalled the warpctc, and it worked like a charm.

**Comment by wuliebucha** [comment]
I have the same problem，conda environment(cudatoolkit-9.0、cudnn-7.1.2、pytorch-0.4.0、warpctc-pytorch-0.1、python-3.6.2 ) . I  doubt it caused by environment, then i tried many version of toolkit, such python( 3.5、3.6) 、cuda(8、9)、cudnn（6、7）、pytorch（0.4、0.41、1.0）and so on, but still can't resolve it, any one have ideas？

**Comment by niuwagege** [comment]
Maybe you can try [this](https://github.com/jpuigcerver/pytorch-baidu-ctc). It works for me!

**Comment by kokoyy** [comment]
> Maybe you can try [this](https://github.com/jpuigcerver/pytorch-baidu-ctc). It works for me!

Thanks, it works for me.
